### PR TITLE
[DOCS-553] DogStatsD mapper

### PR DESCRIPTION
**Pull request by l0k0ms**
### What does this PR do?

Adds DogStatsD mapper documentation.

### Motivation

It has been released with Agent 7.17+

### Preview link

* https://docs-staging.datadoghq.com/gus/dogstatsd-mapper/developers/dogstatsd/dogstatsd_mapper/

### CODE REVIEW

File: content/en/developers/dogstatsd/dogstatsd_mapper.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+| `<METRIC_TO_MATCH>`     | The metric name to match. [Wildcard](#wildcard-match-pattern) or [Regex](#regex-match-pattern) are supported.                             |
+| `<MATCH_TYPE>`          | The type of match to apply to the `<METRIC_TO_MATCH>`. Either [`wildcard`](#wildcard-match-pattern) or [`regex`](#regex-match-pattern)    |
+| `<MAPPED_METRIC_NAME>`  | The new metric name to send to Datadog with the tags defined in the same group.                                                           |
+| `<TAG_KEY>`             | The Tag key to associate to the tags collected.                                                                                           |
+| `<TAG_VALUE_TO_EXPAND>` | The tags collected from the `<MATCH_TYPE>` to inline.                                                                                     |
+
+## Wildcard Match Pattern
+
+The wildcard match pattern matches dot-separated metric names using `*` as wildcard. The metric name must follow the [metrics naming convention][2] for the pattern to work.
```
Comment (AlexandreYang) [inline content/en/developers/dogstatsd/dogstatsd_mapper.md]: Actually, it's a bit more strict than that. Only alphanumeric, `.` and `_` are allowed. 

https://github.com/DataDog/datadog-agent/blob/460ad2c13a7e990ec65660f9dc3aec574c0d6d78/pkg/dogstatsd/mapper/mapper.go#L18

File: content/en/developers/dogstatsd/dogstatsd_mapper.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                '<TAG_KEY>': '<TAG_VALUE_TO_EXPAND>'
+```
+
+With the following placeholders:
+
+| Placeholder             |  Definition                                                                                                                               |
+| ----------------------- | ----------------------------------------------------------------------------------------------------------------------------------------- |
+|  `<PROFILE_NAME>`       | A name to give to your mapping rule profile.                                                                                              |
+| `<PROFILE_PREFIX>`      | The metric name prefix associated to this profile.                                                                                        |
+| `<METRIC_TO_MATCH>`     | The metric name to match. [Wildcard](#wildcard-match-pattern) or [Regex](#regex-match-pattern) are supported.                             |
```
Comment (AlexandreYang) [inline content/en/developers/dogstatsd/dogstatsd_mapper.md]: `METRIC_TO_MATCH` is optional, it defaults to wildcard.

File: content/en/developers/dogstatsd/dogstatsd_mapper.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        match_type: wildcard
+        name: custom_metric
+        tags:
+          tag_key_1: '$1'
+          tag_key_2: '$2'
+```
+
+It would send the metric `custom_metric` to Datadog with the tags `tag_key_1:value_1` and `tag_key_2:value_2`.
+
+**Note**: If the incoming metric have already a tag key that matches one defined in the mapping group, no mapping is applied.
```
Comment (AlexandreYang) [inline content/en/developers/dogstatsd/dogstatsd_mapper.md]:
```suggestion
**Note**: If the incoming metric have already tags, no mapping is applied.
```

Comment (AlexandreYang) [reply content/en/developers/dogstatsd/dogstatsd_mapper.md]: ^ that match more actual impl.

Comment (AlexandreYang) [inline content/en/developers/dogstatsd/dogstatsd_mapper.md]: This not apply to both wildcard and regex type.

File: content/en/developers/dogstatsd/dogstatsd_mapper.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            match_type: regex
+            name: custom_metric
+            tags:
+                tag_key_1: '$1'
+                tag_key_2: '$2'
+```
+
+It would send the metric `custom_metric` to Datadog with the tags `tag_key_1:value_1` and `tag_key_2:value.with.dots._2`.
+
+## Catched group in metric name
```
Comment (AlexandreYang) [inline content/en/developers/dogstatsd/dogstatsd_mapper.md]: It seems the title  is not right, this is not related to `cache`.

File: content/en/developers/dogstatsd/dogstatsd_mapper.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+      prefix: '<PROFILE_PREFIX>'
+      mappings:
+          - match: '<METRIC_TO_MATCH>'
+            match_type: '<MATCH_TYPE>'
+            name: '<MAPPED_METRIC_NAME>'
+            tags:
+                '<TAG_KEY>': '<TAG_VALUE_TO_EXPAND>'
+```
+
+With the following placeholders:
```
Comment (AlexandreYang) [inline content/en/developers/dogstatsd/dogstatsd_mapper.md]: I think it might worth indicating for all fields what's required and what's optional. See https://github.com/DataDog/datadog-agent/pull/4471/files#diff-99e505a5e29849d8a342596285e284a1R743-R753

Comment (l0k0ms) [reply content/en/developers/dogstatsd/dogstatsd_mapper.md]: done

File: content/en/developers/dogstatsd/dogstatsd_mapper.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+-   `job_name:<JOB_NAME>`.
+
+To create a mapping rule:
+
+1. [Open your `datadog.yaml` file][1].
+2. Add a [mapping rule configuration block](#mapping-rule-configuration) under the `dogstatsd_mapper_profiles` parameter.
+
+## Mapping rule configuration
+
+A mapping rule block have the following layout:
```
Comment (sarina-dd) [inline content/en/developers/dogstatsd/dogstatsd_mapper.md]:
```suggestion
A mapping rule block has the following layout:
```

File: content/en/developers/dogstatsd/dogstatsd_mapper.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+Groups extracted can then be expanded with the `$n` format e.g. `$1`, `$2`, `$3`... or the `${n}` format e.g. `${1}`, `${2}`, `${3}`, ...
+
+For instance, if you have the metric `custom_metric.value_1.value_2` with the following mapping group configuration:
+
+```yaml
+dogstatsd_mapper_profiles:
+    - name: my_custom_metric_profile
+      prefix: custom_metric
+      mappings:
+          - match: 'custom_metric.*'
```
Comment (AlexandreYang) [inline content/en/developers/dogstatsd/dogstatsd_mapper.md]: Since 2 wildcards needed since you use $1 and $2. I suggest adding `process` or any metric prefix, more realistic.
```suggestion
          - match: 'custom_metric.process.*..*'
```

File: content/en/developers/dogstatsd/dogstatsd_mapper.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+            match_type: wildcard
+            name: custom_metric
```
Comment (AlexandreYang) [inline content/en/developers/dogstatsd/dogstatsd_mapper.md]:
```suggestion
            name: custom_metric.process
```

File: content/en/developers/dogstatsd/dogstatsd_mapper.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+| `<MAPPED_METRIC_NAME>`  | The new metric name to send to Datadog with the tags defined in the same group.                                                           | yes                     |
+| `<TAG_KEY>`             | The Tag key to associate to the tags collected.                                                                                           | no                      |
+| `<TAG_VALUE_TO_EXPAND>` | The tags collected from the `<MATCH_TYPE>` to inline.                                                                                     | no                      |
+
+## Wildcard Match Pattern
+
+The wildcard match pattern matches dot-separated metric names using `*` as wildcard. The metric name must be only composed of alphanumeric, `.`, and `_` characters for this pattern to work.
+Groups extracted can then be expanded with the `$n` format e.g. `$1`, `$2`, `$3`... or the `${n}` format e.g. `${1}`, `${2}`, `${3}`, ...
+
+For instance, if you have the metric `custom_metric.value_1.value_2` with the following mapping group configuration:
```
Comment (AlexandreYang) [inline content/en/developers/dogstatsd/dogstatsd_mapper.md]:
```suggestion
For instance, if you have the metric `custom_metric.process.value_1.value_2` with the following mapping group configuration:
```

File: content/en/developers/dogstatsd/dogstatsd_mapper.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+      mappings:
+          - match: 'custom_metric.*'
+            match_type: wildcard
+            name: custom_metric
+            tags:
+                tag_key_1: '$1'
+                tag_key_2: '$2'
+```
+
+It would send the metric `custom_metric` to Datadog with the tags `tag_key_1:value_1` and `tag_key_2:value_2`.
```
Comment (AlexandreYang) [inline content/en/developers/dogstatsd/dogstatsd_mapper.md]:
```suggestion
It would send the metric `custom_metric.process` to Datadog with the tags `tag_key_1:value_1` and `tag_key_2:value_2`.
```

File: content/en/developers/dogstatsd/dogstatsd_mapper.md [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+Groups extracted can then be expanded with the `$n` format e.g. `$1`, `$2`, `$3`... or the `${n}` format e.g. `${1}`, `${2}`, `${3}`, ...
+
+For instance, if you have the metric `custom_metric.value_1.value.with.dots._2` with the following mapping group configuration:
+
+```yaml
+dogstatsd_mapper_profiles:
+    - name: my_custom_metric_profile
+      prefix: custom_metric
+      mappings:
+          - match: 'custom_metric\.([\w_]+)\.(.+)'
```
Comment (AlexandreYang) [inline content/en/developers/dogstatsd/dogstatsd_mapper.md]: Same here, I think using something with more than just the prefix is better e.g. `custom_metric.process`  (might need update of other occurrences too)
